src/__experimental__/components/numeral-date/numeral-date.component.js

```diff
@@ -37,86 +53,91 @@ const NumeralDate = ({
     }
   };
 
-  const handleChange = (e, itemId) => {
-    if (onChange) {
-      onChange(e);
-    }
-    if (e.target.value !== dateValue[itemId] && e.target.value.length <= itemId.length) {
-      setDateValue({ ...dateValue, [itemId]: e.target.value });
-    }
-  };
-
-  const handleOnFocus = () => {
-    setIsActive(true);
-  };
+  const handleChange = (e, datePart) => {
+    const { value: newValue } = e.target;
 
-  const handleBlur = (ev) => {
-    const targetObject = {
-      name: ev.target.name,
-      id: ev.target.id,
-      value: { ...dateValue }
-    };
+    if (newValue.length <= datePart.length) {
+      const newDateValue = { ...dateValue, [datePart]: newValue };
+      setDateValue(newDateValue);
 
-    setIsActive(false);
+      /* istanbul ignore else */
```

Comment:
IMO we shouldn't be using `/* istanbul ignore else */` because all tests will still pass if we remove the `if` condition.

Reply:
On the other hand we have to write dummy tests like 'does nothing if onChange is not provided'